satpy/readers/glm_l2.py

```diff
@@ -87,16 +102,23 @@ def get_dataset(self, key, info):
         for attr in ('scene_abbr', 'scan_mode', 'platform_shortname'):
             res.attrs[attr] = self.filename_info[attr]
         # copy global attributes to metadata
-        for attr in ('scene_id', 'orbital_slot', 'instrument_ID', 'production_site', 'timeline_ID'):
+        for attr in ('scene_id', 'orbital_slot', 'instrument_ID',
+                     'production_site', 'timeline_ID', 'spatial_resolution'):
             res.attrs[attr] = self.nc.attrs.get(attr)
         return res
 
+    def _is_2d_xy_var(self, data_arr):
+        has_x_dim = 'x' in data_arr.dims
+        has_y_dim = 'y' in data_arr.dims
+        return has_x_dim and has_y_dim
```

Comment:
This doesn't actually check the number of dimensions. Maybe it could be renamed to `_has_xy_coords` or something?

Reply:
Added a check for the number of dims.